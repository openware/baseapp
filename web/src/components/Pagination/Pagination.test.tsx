import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Pagination } from '.';

const defaultProps = {
    firstElemIndex: 1,
    lastElemIndex: 6,
    total: 30,
    page: 0,
    nextPageExists: true,
    onClickPrevPage: jest.fn(),
    onClickNextPage: jest.fn(),
};

const renderComponent = (props = {}) => render(<Pagination {...{ ...defaultProps, ...props }} />);

describe('Pagination', () => {
    it('should matches snapshot', () => {
        expect(renderComponent().container).toMatchSnapshot();
    });

    it('should have correct className', () => {
        expect(renderComponent().container.querySelector('.pg-history-elem__pagination')).toBeInTheDocument();
    });

    it('has pagination info with right text', () => {
        const { container } = renderComponent();
        expect(container.getElementsByTagName('p')[0].textContent).toEqual('1 - 6 of 30');
    });

    it('should test click on prev page', () => {
        const spyClickPrev = jest.fn();
        const { rerender } = renderComponent({ onClickPrevPage: spyClickPrev });
        const prevButton = screen.getAllByRole('button')[0];

        fireEvent.click(screen.getAllByRole('button')[0]);
        expect(spyClickPrev).toHaveBeenCalledTimes(0);

        rerender(<Pagination {...{ ...defaultProps, onClickPrevPage: spyClickPrev, page: 1 }} />);
        fireEvent.click(prevButton);
        expect(spyClickPrev).toHaveBeenCalledTimes(1);
    });

    it('should test click on next page', () => {
        const spyClickNext = jest.fn();
        const { rerender } = renderComponent({ onClickNextPage: spyClickNext, nextPageExists: false });
        const nextButton = screen.getAllByRole('button')[1];

        fireEvent.click(nextButton);
        expect(spyClickNext).toHaveBeenCalledTimes(0);

        rerender(<Pagination {...{ ...defaultProps, onClickNextPage: spyClickNext }} />);
        fireEvent.click(nextButton);
        expect(spyClickNext).toHaveBeenCalledTimes(1);
    });
});
