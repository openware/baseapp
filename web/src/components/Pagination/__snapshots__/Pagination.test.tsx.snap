// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should matches snapshot 1`] = `
<div
  className="pg-history-elem__pagination"
>
  <p>
    <span>
      1
    </span>
    <span>
       - 
    </span>
    <span>
      6
    </span>
    <span>
       of 
    </span>
    <span>
      30
    </span>
  </p>
  <button
    className="pg-history__pagination-prev"
    disabled={true}
    onClick={[Function]}
  >
    <PreviousIcon
      disabled={true}
    />
  </button>
  <button
    className="pg-history__pagination-next"
    disabled={false}
    onClick={[Function]}
  >
    <NextPageIcon
      disabled={false}
    />
  </button>
</div>
`;
